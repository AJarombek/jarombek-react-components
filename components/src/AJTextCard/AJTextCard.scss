@import "../global";

.aj-text-card {
  @include RobotoThin;
  display: inline-block;
  min-width: 300px;
  height: 500px;
  padding: 5px 10px;
  border-radius: 3px;
  box-shadow: 0 1px 3px rgba(170, 170, 170, 0.5);
}
