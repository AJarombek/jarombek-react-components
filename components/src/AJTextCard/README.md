### Overview

Reusable card component with a header, body, and footer.

**Use cases:**

```javascript
import { AJTextCard } from 'jarombek-react-components';

const CustomComponent = () => 
    <AJTextCard
      title="Sunday, March 22nd"
      subtitle={<>2nd to last Sunday in March</>}
      content={<>Trying to get a little work done today, watch some old basketball games on tv.</>}
      action={() => {}}
      actionText="More Info"
      actionDisabled={false}
    />
```

### Files

| Filename                   | Description                                                                    |
|----------------------------|--------------------------------------------------------------------------------|
| `AJTextCard.js`            | The reusable `AJTextCard` component.                                           |
| `index.js`                 | Index file which exports the `AJTextCard` component.                           |
| `styles.js`                | JSS styles for the `AJTextCard` component.                                     |
