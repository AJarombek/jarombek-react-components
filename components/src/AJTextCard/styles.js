/**
 * Stylesheet for the AJTextCard component using JSS.
 * Author: Andrew Jarombek
 * Date: 12/30/2019
 */

import globalStyles from '../globalStyles';

export default {
  ajTextCard: {
    ...globalStyles.robotoThin,
    display: 'inline-block',
    width: 'auto',
    height: '500px',
    padding: '5px 10px',
    borderRadius: '3px',
    boxShadow: '0 1px 3px rgba(170, 170, 170, 0.5)',

    '&:hover': {
      boxShadow: '0 1px 3px rgba(170, 170, 170, 0.8)',
    },
  },
};
