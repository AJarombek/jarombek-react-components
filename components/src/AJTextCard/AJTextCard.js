/**
 * Reusable generic card component which displays text.
 * @author Andrew Jarombek
 * @since 11/16/2019
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';

import AJTextButton from '../AJTextButton/AJTextButton';
import styles from './styles';

const useStyles = createUseStyles(styles);

/**
 * Component representing a card that displays text and has an action button.
 * @param title Main title of the card.  Can be text or additional React elements.
 * @param subtitle Subtitle of the card.  Can be text or additional React elements.
 * @param content Text content of the card.  Can be text or additional React elements.
 * @param action Function that is called when the action button is clicked.
 * @param actionText Text of the action button in the footer of the card.  Can be text or
 * additional React elements.
 * @param actionDisabled Whether the action button is disabled from being clicked or not.
 * @return {*} React elements that make up a text card.
 */
const AJTextCard = ({title, subtitle, content, action, actionText, actionDisabled=false}) => {
  const classes = useStyles();

  return (
    <div className={classes.ajTextCard}>
      <div className="aj-text-card-header">
        <h2 className="aj-text-card-title">{title}</h2>
        <p className="aj-text-card-subtitle">{subtitle}</p>
      </div>
      <div className="aj-text-card-body">
        <p className="aj-text-card-content">{content}</p>
      </div>
      <div className="aj-text-card-footer">
        <AJTextButton onClick={action} disabled={actionDisabled}>{actionText}</AJTextButton>
      </div>
    </div>
  );
};

AJTextCard.propTypes = {
  title: PropTypes.node.isRequired,
  subtitle: PropTypes.node,
  content: PropTypes.node.isRequired,
  action: PropTypes.func.isRequired,
  actionText: PropTypes.node.isRequired,
  actionDisabled: PropTypes.bool
};

export default AJTextCard;
