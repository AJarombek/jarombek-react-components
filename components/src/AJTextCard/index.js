import AJTextCard from './AJTextCard';

export default AJTextCard;
