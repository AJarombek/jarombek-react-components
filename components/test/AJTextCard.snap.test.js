import React from 'react';
import AJTextCard from '../src/AJTextCard/AJTextCard';
import renderer from 'react-test-renderer';

/**
 * Snapshot test for the {@link ../../components/AJTextCard} component.
 * @author Andrew Jarombek
 * @since 12/19/2019
 */

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AJTextCard
        title="Snapshot Test"
        content={<div>Hello from snapshot test</div>}
        actionText="Submit"
        action={(f) => f}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const tree = renderer
    .create(
      <AJTextCard
        title="Snapshot Test 2"
        subtitle="Test with all props"
        content={<div>Hello from the second snapshot test</div>}
        actionDisabled={true}
        actionText="Submit"
        action={(f) => f}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
