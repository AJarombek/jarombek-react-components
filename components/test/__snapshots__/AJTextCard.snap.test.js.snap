// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajTextCard-0-2-1 aj-text-card"
>
  <div
    className="aj-text-card-header"
  >
    <h2
      className="aj-text-card-title"
    >
      Snapshot Test
    </h2>
    <p
      className="aj-text-card-subtitle"
    />
  </div>
  <div
    className="aj-text-card-body"
  >
    <p
      className="aj-text-card-content"
    >
      <div>
        Hello from snapshot test
      </div>
    </p>
  </div>
  <div
    className="aj-text-card-footer"
  >
    <div
      className="ajBaseButton-0-2-5 ajBaseButtonEnabled-0-2-6 aj-text-button ajTextButton-0-2-2 ajTextButtonEnabled-0-2-3"
      onClick={[Function]}
    >
      <button
        disabled={false}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly with all props 1`] = `
<div
  className="ajTextCard-0-2-1 aj-text-card"
>
  <div
    className="aj-text-card-header"
  >
    <h2
      className="aj-text-card-title"
    >
      Snapshot Test 2
    </h2>
    <p
      className="aj-text-card-subtitle"
    >
      Test with all props
    </p>
  </div>
  <div
    className="aj-text-card-body"
  >
    <p
      className="aj-text-card-content"
    >
      <div>
        Hello from the second snapshot test
      </div>
    </p>
  </div>
  <div
    className="aj-text-card-footer"
  >
    <div
      className="ajBaseButton-0-2-5 ajBaseButtonDisabled-0-2-7 aj-text-button ajTextButton-0-2-2 ajTextButtonDisabled-0-2-4"
      onClick={null}
    >
      <button
        disabled={true}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
